### PR TITLE
Document+improve serializers

### PR DESCRIPTION
**Pull request by StachuDotNet**

### CODE REVIEW

File: fsharp-backend/src/LibExecution/OCamlTypes.fs [hunk 1]
```diff
@@ -232,7 +232,7 @@ module RuntimeT =
     | DResp of (dhttp * dval)
     | DDB of string
     | DDate of RT.DDateTime.T
-    | DPassword of Password // We dont use this path for testing, see DvalReprExternal.Tests
+    | DPassword of Password // We dont use this path for testing, see DvalReprLegacyExternal.Tests
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/OCamlTypes.fs:235]:
```suggestion
    | DPassword of Password // We dont use this path for testing, see DvalRepr.Tests
```

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 2]
```diff
@@ -15,17 +15,12 @@ module LibExecution.DvalReprExternal
 // here.
 
 open Prelude
-open VendoredTablecloth
+open Tablecloth
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:18]:
```suggestion
open VendoredTablecloth
```

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 3]
```diff
@@ -123,7 +117,7 @@ let ocamlStringOfFloat (f : float) : string =
   // format "%.12g".
   // https://github.com/ocaml/ocaml/blob/4.07/stdlib/stdlib.ml#L274
 
-  // CLEANUP We should move on to a nicer format. See DvalReprExternal.tests for edge cases. See:
+  // CLEANUP We should move on to a nicer format. See DvalReprLegacyExternal.tests for edge cases. See:
```
Comment (StachuDotNet) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:120]:
```suggestion
  // CLEANUP We should move on to a nicer format. See DvalRepr.Tests for edge cases. See:
```

**Comment by StachuDotNet** [comment]
I've just reviewed this once over; looks good to me.

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /// Ways of converting Dvals to/from strings, with external use allowed
-module LibExecution.DvalReprExternal
+module LibExecution.DvalReprLegacyExternal
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:2]: I think the comments here could start with something like:

"The remaining functions in this file are used by the HttpClient or the Http framework, and cannot be changed in their existing usages. The plan to remove them is to replace both the Http framework with a middleware based design, with some of that middleware being a typed json serializer. The typed json serializer would be a standard library function, not something that's an inate part of the framework/client in the way that these functions are"

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 5]
```diff
@@ -139,71 +133,12 @@ let ocamlStringOfFloat (f : float) : string =
...
-
-// When printing to grand-users (our users' users) using text/plain, print a
-// human-readable format. TODO: this should probably be part of the functions
-// generating the responses. Redacts passwords.
-
+// SERIALIZER_DEF DvalReprLegacyExternal.toEnduserReadableTextV0
+// Plan: We'd like to deprecate this in favor of an improved version only
+// usable/used by StdLib functions in various http clients and middlewares.
+/// When printing to grand-users (our users' users) using text/plain, print a
+/// human-readable format. TODO: this should probably be part of the functions
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:140]: > TODO: this should probably be part of the functions
/// generating the responses.

I don't know what that says

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 6]
```diff
@@ -284,11 +220,15 @@ let toEnduserReadableTextV0 (dval : Dval) : string =
 
   reprfn dval
 
+// SERIALIZER_DEF STJ DvalReprLegacyExternal.toPrettyMachineJsonV1
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:223]: This is confusing, as the situation for this is the same as the other functions in this file.

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs [hunk 7]
```diff
@@ -367,74 +306,8 @@ let toPrettyMachineJsonStringV1 (dval : Dval) : string =
...
-    | DOption (Some dv) -> "Just " + toRepr_ indent dv
-    | DResult (Ok dv) -> "Ok " + toRepr_ indent dv
-    | DResult (Error dv) -> "Error " + toRepr_ indent dv
-    | DErrorRail dv -> "ErrorRail: " + toRepr_ indent dv
-    | DBytes bytes -> Base64.defaultEncodeToString bytes
-
-  toRepr_ 0 dv
-
-
+// SERIALIZER_DEF DvalReprLegacyExternal.unsafeOfUnknownJsonV0
```
Comment (pbiggar) [inline fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:309]: Again, the situation here is the same as in the other ones, but the comment is wildly different.

**Review by pbiggar — CHANGES_REQUESTED** [review]
Looks good, ready to merge with the stated changes.

**Comment by pbiggar** [comment]
I'm building on top of this - could you merge?

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs (hunk 5 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs]: Nor do I - removed it.

File: fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs (hunk 6 above)
Comment (StachuDotNet) [reply fsharp-backend/src/LibExecution/DvalReprLegacyExternal.fs:230]: I'm really not sure what you mean by this - maybe I've just forgotten the context. I've added an inline TODO to revise the commentary.

**Comment by pbiggar** [comment]
Thanks!
